hs/src/Reach/Eval/Core.hs

```diff
@@ -2734,18 +2750,18 @@ evalPrim p sargs =
       let existingToks = st_toks st
       tokIsUniq <- tokIsUnique (map DLA_Var existingToks) $ DLA_Var tokdv
       doClaim CT_Assume tokIsUniq $ Just "New token is unique"
-      setSt $
-        st
-          { st_toks = existingToks <> [tokdv]
-          , st_toks_c = S.insert tokdv (st_toks_c st)
-          , st_tok_pos = M.insert (DLA_Var tokdv) (length existingToks) (st_tok_pos st)
-          }
+      setSt $ st { st_toks_c = S.insert tokdv (st_toks_c st) }
       let toka = DLA_Var tokdv
-      let mtok_a = Just $ toka
-      doBalanceInit' TM_Balance mtok_a supplya
+      trackToken toka supplya
       tokenMetaSet TM_Supply toka supplya False
       tokenMetaSet TM_Destroyed toka (DLA_Literal $ DLL_Bool False) False
       return $ public $ SLV_DLVar tokdv
+    SLPrim_Token_track -> do
```

Comment:
You need to call this in the `publish` code path that currently modifies the set of tokens, so we can delete code by doing this